File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <deque>

using namespace std;

int main(void) {
    deque<int> d;
    int n;
    
    cin >> n;
    for (int i = 0; i < n; i++) {
        string command;
        
        cin >> command;
        
        if (command == "push_front") {
            int number;
            cin >> number;
            
            d.push_front(number);
        } else if (command == "push_back") {
            int number;
            cin >> number;
            
            d.push_back(number);
        } else if (command == "pop_front") {
            if (d.empty()) {
                cout << -1 << '\n';
            } else {
                cout << d.front() << '\n';
                d.pop_front();
            }            
        } else if (command == "pop_back") {
            if (d.empty()) {
                cout << -1 << '\n';
            } else {
                cout << d.back() << '\n';
                d.pop_back();
            }
        } else if (command == "size") {
            cout << d.size() << '\n';
        } else if (command == "empty") {
            if (d.empty()) {
                cout << 1 << '\n';
            } else {
                cout << 0 << '\n';
            }
        } else if (command == "front") {
            if (d.empty()) {
                cout << -1 << '\n';
            } else {
                cout << d[0] << '\n';
            }
        } else if (command == "back") {
            if (d.empty()) {
                cout << -1 << '\n';
            } else {
                cout << d[d.size() - 1] << '\n';
            }
        }
    }
    return 0;
}

File: 백준/Silver/10866. 덱/덱.py
from collections import deque
import sys

N = int(input())
d = deque()

for _ in range(N) :
    command = sys.stdin.readline().split()
    
    if (command[0] == "push_front") :
        d.appendleft(command[1])
    elif (command[0] == "push_back") :
        d.append(command[1])
    elif (command[0] == "pop_front") :
        if (len(d) == 0) :
            print(-1)
        else :
            print(d[0])
            d.popleft()
    elif (command[0] == "pop_back") :
        if (len(d) == 0) :
            print(-1)
        else :
            print(d[len(d) - 1])
            d.pop()
    elif (command[0] == "size") :
        print(len(d))
    elif (command[0] == "empty") :
        if (len(d) == 0) :
            print(1)
        else :
            print(0)
    elif (command[0] == "front") :
        if (len(d) == 0) :
            print(-1)
        else :
            print(d[0])
    elif (command[0] == "back") :
        if (len(d) == 0) :
            print(-1)
        else :
            print(d[len(d) - 1])